HW.rb

```diff
@@ -0,0 +1,213 @@
+        name: "Ashjan",
+        transactions: [
+            {
+                type: "COFFEE",
+                amount: 7.43
+            },
+            {
+                type: "TACOS",
+                amount: 14.65
+            },
+            {
+                type: "COFFEE",
+                amount: 4.43
+            }
+        ]
+    },
+    {
+        name: "Abeer",
+        transactions: [
+            {
+                type: "BIKES",
+                amount: 800.00
+            },
+            {
+                type: "TACOS",
+                amount: 14.65
+            },
+            {
+                type: "COFFEE",
+                amount: 4.43
+            }
+        ]
+    },
+    {
+        name: "Nujud",
+        transactions: [
+            {
+                type: "COFFEE",
+                amount: 7.43
+            },
+            {
+                type: "COFFEE",
+                amount: 100.00
+            },
+            {
+                type: "COFFEE",
+                amount: 4.43
+            }
+        ]
+    }
+  ]
+
+  coffee_average_per_person = []
+  for person in people do
+    sum = 0
+    num = 0
+      for amount in person[:transactions] do
+          if amount[:type] == "COFFEE"
+            sum += amount[:amount] 
+                num += 1
+          end
+      end
+     avarge = sum / num 
+    coffee_average_per_person.push({name: person[:name], coffee_average: avarge})
+end
+puts "Question 3"
+p coffee_average_per_person
+
+  # Question 4
+  stores = [
+    {
+        store_name: "Virgin",
+        products: [
+            {
+                description: "PS4",
+                price: 9384.33
+            },
+            {
+                description: "Pen",
+                price: 345.54
+            }
+        ]
+    },
+    {
+        store_name: "Manuel",
+        products: [
+            {
+                description: "Bread",
+                price: 654.44
+            },
+            {
+                description: "Cheese",
+                price: 323.43
+            }
+        ]
+    },
+    {
+        store_name: "Noon",
+        products: [
+            {
+                description: "iPhone",
+                price: 345.43
+            },
+            {
+                description: "TV",
+                price: 899.33
+            }
+        ]
+    }
+  ]
+  most_expensive_products_by_store = []
+
+  for store in stores do
+    price = 0
+    most = {}
+    for product in store[:products] do
+        if product[:price] > price
+            price = product[:price]
+            most = product
+        end
+    end
+    most_expensive_products_by_store.push(store_name:store[:store_name],  most_expensive_product: most)
+  end
+  puts "Question 4"
+  p most_expensive_products_by_store
+
+  # Bonus
+  puts "Bonus"
+  
+  friend = " "
+friends = []
+while friend != "n" do
+    p "add a student" 
+    friend = gets.chomp
+    friends.push(friend)
+    p "Do you want to continue ? (y/n)" 
```

Comment:
what happend if i type some letter not Y or N ?

Reply:
mmm I tried to do it like this ... 
while friend != "n" || friend != "No" || friend != "no" ||friend != "NO"
but unfortunately it doesn't work probably with me